tests/certification/bindings/cron/cron_test.go

```diff
@@ -0,0 +1,223 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cron_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/dapr/components-contrib/bindings"
+	"github.com/dapr/components-contrib/bindings/cron"
+	bindings_loader "github.com/dapr/dapr/pkg/components/bindings"
+
+	"github.com/dapr/dapr/pkg/runtime"
+	"github.com/dapr/kit/logger"
+
+	dapr_testing "github.com/dapr/dapr/pkg/testing"
+	"github.com/stretchr/testify/require"
+
+	daprClient "github.com/dapr/go-sdk/client"
+	"github.com/dapr/go-sdk/service/common"
+
+	// Certification testing runnables
+	"github.com/dapr/components-contrib/tests/certification/embedded"
+	"github.com/dapr/components-contrib/tests/certification/flow"
+	"github.com/dapr/components-contrib/tests/certification/flow/app"
+	"github.com/dapr/components-contrib/tests/certification/flow/sidecar"
+	"github.com/dapr/components-contrib/tests/certification/flow/watcher"
+)
+
+// Test cron trigger with most frequent schedule possible : @every 1s with a context deadline
+func TestCronBindingFrequentTrigger(t *testing.T) {
+	ports, _ := dapr_testing.GetFreePorts(3)
+	grpcPort := ports[0]
+	httpPort := ports[1]
+	appPort := ports[2]
+
+	triggerWatcher := watcher.NewOrdered()
+	for i := 1; i <= 10; i++ {
+		triggerWatcher.ExpectInts(i)
+	}
+
+	// test if cron triggers 10 times within 10 seconds
+	testAssert := func(ctx flow.Context) error {
+		triggerWatcher.Assert(ctx, time.Second*10)
+		return nil
+	}
+
+	application := func(ctx flow.Context, s common.Service) error {
+		triggeredCount := 0
+		// Setup the input binding endpoint
+		err := s.AddBindingInvocationHandler("cron", func(_ context.Context, in *common.BindingEvent) ([]byte, error) {
+			ctx.Logf("Cron triggered at %s", time.Now().String())
+			triggeredCount++
+			triggerWatcher.Observe(triggeredCount)
+			return []byte("{}"), nil
+		})
+		require.NoError(t, err)
+		return nil
+	}
+
+	testDelete := func(ctx flow.Context) error {
+		client := sidecar.GetClient(ctx, "cron-sidecar")
+
+		req := &daprClient.InvokeBindingRequest{Name: "cron", Operation: "delete"}
+		res, err := client.InvokeBinding(ctx, req)
+
+		require.Equal(t, res.Metadata["schedule"], "@every 1s")
+		require.NotEmpty(t, res.Metadata["stopTimeUTC"])
+		require.NoError(t, err, "Error cancelling cron schedule")
+		return nil
+	}
+
+	flow.New(t, "test cron trigger schedule @every 1s").
+		Step(app.Run("cronapp", fmt.Sprintf(":%d", appPort), application)).
+		Step(sidecar.Run("cron-sidecar",
+			embedded.WithComponentsPath("./components"),
+			embedded.WithDaprGRPCPort(grpcPort),
+			embedded.WithAppProtocol(runtime.HTTPProtocol, appPort),
+			embedded.WithDaprHTTPPort(httpPort),
+			componentRuntimeOptions(),
+		)).
+		Step("assert cron triggered within deadline", testAssert).
+		Step("test trigger delete operation", testDelete).
+		Step("stop sidecar", sidecar.Stop("cron-sidecar")).
```

Comment:
Once deleted, should we also check if cron was not triggered again?

Reply:
Yes we should.

Reply:
The delete operation is broken for latest dapr.
This line never gets called when the runningCtx is cancelled
https://github.com/dapr/components-contrib/blob/ede060959a0a1deb1e7980aac6a4f5199b31e6a2/bindings/cron/cron.go#L91
While calling `b.Read()` and `b.Invoke()`, `b` is different since cron is both input/output binding.
Hence, reverting to using `stopCh map[string]chan bool` to signal ctx cancellation and stopping cron.